### PR TITLE
Online: rename _facts -> _info

### LINKED ISSUE #60517: Online cloud _facts modules need to be renamed to _info and stop returning ansible_facts
##### SUMMARY
According to the [requirements](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements), `_facts` modules should return information that is specific to the host the module runs on as `ansible_facts`. (See also ansible/ansible#54280 for some discussion.)

I think that the following modules do not satisfy this condition: they return information that is not specific to the host machine, but bound to a cloud account. Please verify whether this is actually the case.

To fix this, the core team thinks the best approach is to rename the `_facts` modules to `_info` modules, and change the return value so that if the module is called with the `_facts` name (should be deprecated, and will then be removed in Ansible 2.13), it should return `ansible_facts` as now (and print a deprecation message), and if it is called with the `_info` name, it no longer returns `ansible_facts`, so that the result needs to be registered. This informs users that they have to change something, won't break their playbooks, and allows them (by switching to the `_info` names) to verify that their roles and playbooks no longer use the facts.

If that approach doesn't work, or you want to change the module's interface (like rename the return value(s)), it is better to deprecate the `_facts` module and create a new `_info` module next to it. To see both approaches implemented, you can take a look at https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f (ansible/ansible#60178).

If you have any input — preferred method to solve this, doubts that all modules should be renamed, ... — please write it here!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
online_server_facts
online_user_facts

##### ANSIBLE VERSION
```paste below
2.9.0
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #60517.

##### ISSUE TYPE
- Bugfix Pull Request
- New Module Pull Request

##### COMPONENT NAME
online_server_facts
online_user_facts

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

**Comment by felixfontein** [comment]
Rebased to resolve conflict in porting guide.

**Comment by felixfontein** [comment]
CC @sieben (module author)

File: lib/ansible/modules/cloud/online/online_server_info.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  - U(https://www.online.net/en/dedicated-server)
+version_added: "2.9"
+author:
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: online
+'''
+
+EXAMPLES = r'''
+- name: Gather Online server information
+  online_server_facts:
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/online/online_server_info.py]:
```suggestion
  online_server_info:
```

File: test/legacy/roles/online_server_facts/tasks/main.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
-# ONLINE_TOKEN='XXX' ansible-playbook ./test/legacy/online.yml --tags test_online_server_facts
```
Comment (s-hertel) [inline test/legacy/roles/online_server_facts/tasks/main.yml:1]: Also need to rename the directories test/legacy/roles/online_server_facts and test/legacy/roles/online_user_facts so the roles in test/legacy/online.yml work.

**Review by s-hertel — CHANGES_REQUESTED** [review]
A couple little things, but looks good besides that.

File: test/legacy/roles/online_server_facts/tasks/main.yml (hunk 2 above)
Comment (felixfontein) [reply test/legacy/roles/online_server_facts/tasks/main.yml:1]: Looks like I didn't notice when editing the file... Should have also run `find | grep xxx` here... Thanks!

File: lib/ansible/modules/cloud/online/online_server_info.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/online/online_server_info.py]: Thanks!

**Review by s-hertel — APPROVED** [review]

**Comment by felixfontein** [comment]
@s-hertel thanks a lot for reviewing this one, too!

**Pull request merged**